mlir/utils/jenkins/Jenkinsfile.upstream

```diff
@@ -21,6 +21,7 @@ pipeline {
                 git 'https://github.com/llvm/llvm-project'
                 cmakeBuild generator: 'Ninja',\
                     buildDir: 'build',\
+                    sourceDir: 'llvm',\
```

Comment:
Why does this `sourceDir` matter? Why is it not `mlir`?

Reply:
`sourceDir` here is the directory containing the CMakeLists.txt, and, in the previous Jenkinsfile, we built from `llvm` (and that's what our current setup also does - it builds `llvm`, which then goes on to include `mlir` under `tools/`)

Reply:
I'm not convinced this does anything. Taking a look back, I think that `../llvm` is added by mistake and you can build with `..` all the same - We do also have a `CMakeLists.txt` in the root directory.

Since we are not using this jenkinsfile anyways, I'm fine with you keep this as-is.

Reply:
If you look at the context, this isn't our build - this is a build of LLVM from https://github.com/llvm/llvm-project - which doesn't have a CMakeLists.txt in the root.

However, if I misunderstood the setup and the idea is to test our code but with upstream's files, then I can change it.

Reply:
> this is a build of LLVM from

You get the idea right. Though, I still remember when I used to compile the tip of upstream llvm and don't need to do `../llvm` in the configuration step. 

I think this is an unimportant implementation detail that can always get verified easily when we start to build upstream tip again. I'm marking this as resolved.